src/border_agent/border_agent.cpp

```diff
@@ -251,6 +254,13 @@ void BorderAgent::Start(void)
     UpdateMeshCopService();
 
     otBorderAgentSetEphemeralKeyCallback(mHost.GetInstance(), BorderAgent::HandleEpskcStateChanged, this);
+
+#if OTBR_ENABLE_FEATURE_FLAGS
+    FeatureFlagList featureFlagList;
```

Comment:
I noticed that we create a local variable of `FeatureFlagList` type here.  Does the `FeatureFlagList` constructor properly initialize it? How does it get set?

This might be unrelated to the changes in this PR, and more about how `FeatureFlagList` is intended to work. Is this the intended usage, with different modules creating their own instances of `FeatureFlagList`? Should we consider making it a singleton instance instead?

Reply:
+1 on Abtin's concern. 

I think we should keep using a centralized place to initialize/apply the feature flags instead of creating a new feature flag instance here.

Currently the feature flag's input is from dbus:
https://github.com/openthread/ot-br-posix/blob/f2fcf8b5b8564bd42c36c249b2d3f250e00802c2/src/dbus/server/dbus_thread_object_rcp.cpp#L1239

Here's how it's takes effect: https://github.com/openthread/ot-br-posix/blob/f2fcf8b5b8564bd42c36c249b2d3f250e00802c2/src/ncp/rcp_host.cpp#L252

Reply:
Thanks for input! We have a discussion on how to handle featureflaglist in OTBR ePSKC feature; summary: would not use featureflag(default) in init, ePSKc has it's own init vale; keep current method for applyFeature list, and refactor to callback function(tracking [issue]([b/358293040](http://b/358293040))) later. [Note](https://docs.google.com/document/d/1m_-2xhL43CaMNfa10nWDdxrd1X_YUbdLQUD9UUxfIJo/edit?usp=sharing&resourcekey=0-40QZF81WWOKyQtWoWfM8tA)